### PR TITLE
Fix lint_and_format action so lint findings are not ignored

### PR DESCRIPTION
**Pull request by andrewpollock**
Fixes: #521

### CODE REVIEW

**Review by oliverchang — APPROVED** [review]
thanks!

File: tools/lint_and_format.sh [hunk 1]
```diff
@@ -14,5 +14,19 @@
 # limitations under the License.
 
 script_dir=$(dirname "$0")
-git ls-files | grep '\.py$' | grep -v '_pb2' | grep -v 'third_party' | xargs pylint --rcfile="$script_dir/../.pylintrc"
-git ls-files | grep '\.py$' | grep -v '_pb2' | grep -v 'third_party' | xargs yapf -d --style "$script_dir/../.style.yapf"
+IN_SCOPE_FILES="$(git ls-files | grep '\.py$' | grep -v -E '(_pb2|third_party)')"
+
+lint_findings=""
+if ! echo "$IN_SCOPE_FILES" | xargs pylint --rcfile="$script_dir/../.pylintrc"; then
+  lint_findings="lint_findings"
```
Comment (oliverchang) [inline tools/lint_and_format.sh:21]: nit: this seems a bit confusing to read at first wrt intention. I thought this was reading the same variable at first.

Maybe just stick with the number here, and use `[ $lint_findings -ne 0 ]` ?

Comment (andrewpollock) [reply tools/lint_and_format.sh:21]: I've done this so it's a little easier to comprehend when running under `sh -x`

You'll see the string "lint_findings" and/or "format_findings" so it's obvious with check has failed here.

Comment (oliverchang) [reply tools/lint_and_format.sh:21]: ack, that's a great point. Let's keep this then!

**Review by oliverchang — APPROVED** [review]

**Pull request merged**